### DISCUSSION TITLE
Looking for projects to test the checker on

### DISCUSSION THREAD

**Discussion by kaleidawave** [kaleidawave/ezno#195] [Announcements]
Hey all 👋 

If you haven't already seen the project has broken the 300 specification test passing mark! Now after adding narrowing (#156) it has shot past that, now passing [323](https://github.com/kaleidawave/ezno/actions/runs/10478894299/job/29023306493#step:7:365) tests passing on the main branch 🎉 

Although the goal of maximising features and type checking correctness is great, it has slightly misaligned the project away from the practical side of actually type checking real world code.

I have a few libraries and existing code that I have been testing with the type checker on, but I am interested to see more!

I have started on PRS #185 and #187 which are aimed in improving modules and collection types respectively. This weekend I am interested in focusing those features on real world code and looking to find things that might be missed in the current specification tests. 

I also am thinking about adding compiler flags to tone down some of the analysis the checker does to increase usability.

Leave them in the comments below and I will try them out this weekend. Also I will try and reply to what things are currently not implemented or working correctly for the checker to correctly check the project. (Don't bother checking them with the latest release as it is a bit outdated now). and no guarantees but there might be chances where I can get quick fixes into the next release (early next week?)

In the future these projects could be used in something like a ecosystem CI check

**Comment by kaleidawave** [comment]
Suggestion from Twitter:

The `@std` library (from Deno)

- https://jsr.io/@std/text looks a good start
- low dependencies

**Comment by kaleidawave** [comment]
Another suggestion from Twitter:

[naiveasync](https://github.com/untra/naiveasync)

- Lots of dependencies including React, but a basic check without events could work!

**Comment by kaleidawave** [comment]
Another suggestion from Twitter:

https://github.com/getify/deePool

- https://github.com/getify/deePool/blob/master/src/deePool.js looks usable, might work without any constraint flags

https://github.com/getify/CAF/blob/master/src/caf.js & https://github.com/getify/fasy/blob/master/src/concurrent.js 

- use `async` and `Promise`, so not sure yet for full

All above use CJS, which isn't implement ATM but I think is quite possible (using constant functions) (something to think about in #185)

**Comment by kaleidawave** [comment]
One of my favourite snippets: https://github.com/cprosche/mulberry32

Also just found this repository: https://github.com/JoakimCh/pluggable-prng should check that all the additions in #186 don't break it

**Comment by kaleidawave** [comment]
Another suggestion from Twitter:

https://github.com/silen-z/fiveway/tree/main/packages/fiveway

**Comment by Aetherall** [comment]
If you want I am making a library that uses very complex types

https://github.com/ddd-ts/monorepo

**Reply by Aetherall** [reply]
I can setup a specific branch that checks using ezno and a "failure allowed" CI job

**Reply by kaleidawave** [reply]
Looks interesting. Is there maybe a single file (with no imports) that has some interesting type code that would be good to test. 

I can try whole repositories if they are simple (not too many files or dependencies or monorepo setups) but single files are easier for now.

**Reply by Aetherall** [reply]
There is this one:
https://github.com/ddd-ts/monorepo/blob/main/packages/freeze/src/divergence.ts
With the "tests"
https://github.com/ddd-ts/monorepo/blob/main/packages/freeze/src/divergence.spec.ts

These two files have no dependencies.
I use that to compute a pretty output for assignation compatibility in one of the tools.


If you want more "class" oriented typing, I also have Trait that have no dependencies.
https://github.com/ddd-ts/monorepo/blob/main/packages/traits/src/index.ts
https://github.com/ddd-ts/monorepo/blob/main/packages/traits/src/index.spec.ts
It is a tool that leverages inheritence chains to provide composition ( i live a dangerous life )

I think they would make a good benchmark

**Reply by kaleidawave** [reply]
Those look interesting definitions to turn into types! When I get the chance will have a look

**Comment by kaleidawave** [comment]
Interesting

https://github.com/bigmistqke/view.gl

I wonder if the checker could do *stuff* around webgl?

**Comment by TylorS** [comment]
Some projects which might be interesting at some point:

- https://github.com/Snowflyt/hkt-core for stress-testing `this` type-checking
- https://github.com/Effect-TS/effect has a lot of interesting types as well, mostly computed/conditional types which derive information from another structure. Many of their packages have `dtslint` tests for typechecking which might be a nice way to gain a bunch of additional type-checking tests

**Reply by kaleidawave** [reply]
Good suggestions! Yeah the type-checking is on the complex end of the spectrum. But maybe good candidates for parsing type-annotations

**Reply by jasikpark** [reply]
Any https://tanstack.com project could be good.
